tests.py

```diff
@@ -472,68 +472,111 @@ def test_csv_download_in__csv_listen_out(self):
+        self.assertTrue(filecmp.cmp(TESTING_IN_FILE_BIN,TESTING_OUT_FILE_BIN))
 
+    def test_csv_download_in__bin_listen_out(self):
         host="localhost"
-        port=5013
+        port=5016
 
         a_pipeline = Pipeline()
         a_file_source = FileSource( filename=TESTING_IN_FILE_BIN,
                                     pipeline=a_pipeline)
-        a_pipeline.add_processing_step(ListenSink(max_receivers=5,host=host,port=port))
+        a_pipeline.add_processing_step(ListenSink(  max_receivers=5,
+                                                    host=host,
+                                                    port=port))
         a_file_source.start()
         a_pipeline.start()
+        b_pipeline = Pipeline()
+        b_download_source = DownloadSource( host=host,
+                                            port=port,
+                                            pipeline=b_pipeline)
+        b_pipeline.add_processing_step(FileSink(filename=TESTING_OUT_FILE_CSV,
+                                                data_format=CSV_DATA_FORMAT_IDENTIFIER))
+        b_pipeline.start()
+        b_download_source.start()
+        time.sleep(self.DEFAULT_SLEEPING_DURATION)
+        a_file_source.stop()
+        a_pipeline.stop()
+        b_download_source.stop()
+        b_pipeline.stop()
 
-        time.sleep(5)
+        self.assertTrue(filecmp.cmp(TESTING_IN_FILE_CSV,TESTING_OUT_FILE_CSV))
 
+    def test_bin_download_in__csv_listen_out(self):
+        host="localhost"
+        port=5017
+        a_pipeline = Pipeline()
+        a_file_source = FileSource( filename=TESTING_IN_FILE_BIN,
+                                    pipeline=a_pipeline)
+        a_pipeline.add_processing_step(ListenSink(  max_receivers=5,
+                                                    host=host,
+                                                    port=port))
+        a_file_source.start()
+        a_pipeline.start()
         b_pipeline = Pipeline()
-        b_download_source = DownloadSource(host=host,port=port,pipeline=b_pipeline)
-        b_pipeline.add_processing_step(FileSink(filename=TESTING_OUT_FILE_CSV))
+        b_download_source = DownloadSource( host=host,
+                                            port=port,
+                                            pipeline=b_pipeline)
+        b_pipeline.add_processing_step(FileSink(filename=TESTING_OUT_FILE_BIN,
+                                                data_format=BINARY_DATA_FORMAT_IDENTIFIER))
         b_pipeline.start()
         b_download_source.start()
-
-        time.sleep(5)
+        time.sleep(self.DEFAULT_SLEEPING_DURATION)
         a_file_source.stop()
         a_pipeline.stop()
-        time.sleep(5)
         b_download_source.stop()
         b_pipeline.stop()
-        time.sleep(5)
 
-        import filecmp
-        self.assertTrue(filecmp.cmp(TESTING_IN_FILE_CSV,TESTING_OUT_FILE_CSV))
+        self.assertTrue(filecmp.cmp(TESTING_IN_FILE_BIN,TESTING_OUT_FILE_BIN))
```

Comment:
There should be a way to compare file contents with a better assertion failure message... At least the files should be loaded, and their contents should be compared with something like assertEquals(str1, str2)